### ISSUE TITLE
cancer detection system

### ISSUE THREAD

**Issue by safwanriyaz** [safwanriyaz/data-science#1]

**Comment by safwanriyaz** [comment]
`import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten,Dense
from tensorflow.keras.preprocessing.image import ImageDataGenerator

train_datagen = ImageDataGenerator(rescale=1./255, rotation_range=20, zoom_range=0.2)
train_generator = train_datagen.flow_from_directory(
    "C:/Data Science/CNN/dataset-2/train", target_size=(150, 150), batch_size=32, class_mode='binary'
)

validation_datagen = ImageDataGenerator(rescale=1./255)
validation_generator = validation_datagen.flow_from_directory(
   "C:/Data Science/CNN/dataset-2/validation" , target_size=(150,150), batch_size=32, class_mode='binary'
)

model = Sequential([
    Conv2D(32, (3, 3), activation='relu', input_shape=(150, 150, 3)),
    MaxPooling2D(pool_size=(2, 2)),
    Conv2D(64, (3, 3), activation='relu'),
    MaxPooling2D(pool_size=(2, 2)),
    Flatten(),
    Dense(128, activation='relu'),
    Dense(1, activation='sigmoid')
])

model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
model.fit(train_generator, epochs=10, validation_data=validation_generator)

model.save('ctscan_cancer_model.h5')





import numpy as np
from tensorflow.keras.models import load_model
from tensorflow.keras.preprocessing.image import load_img,img_to_array

model=load_model('ctscan_cancer_model.h5')
img=load_img("C:/Data Science/CNN/cancer.jpg",target_size=(150,150))
img_array = img_to_array(img) / 255.0
img_array=np.expand_dims(img_array,axis=0)

prediction=model.predict(img_array)
print(f"prediction:{prediction[0][0]:.4f}")`

**Comment by safwanriyaz** [comment]
`import numpy as np
from tensorflow.keras.models import load_model
from tensorflow.keras.preprocessing.image import load_img, img_to_array
import tkinter as tk
from tkinter import filedialog
from PIL import Image, ImageTk
import webbrowser

model = load_model('ctscan_cancer_model.h5')

def predict_cancer(image_path):
    img = load_img(image_path, target_size=(150, 150))  
    img_array = img_to_array(img) / 255.0
    img_array = np.expand_dims(img_array, axis=0)
    prediction = model.predict(img_array)
    probability = prediction[0][0]
    
    if probability >= 0.5:
        diagnosis = f"Prediction: {probability:.4f} - Cancer detected."
        if probability >= 0.8:
            condition = "Condition: Critical."
        elif 0.6 <= probability < 0.8:
            condition = "Condition: Severe."
        else:
            condition = "Condition: Mild."
    else:
        diagnosis = f"Prediction: {probability:.4f} - No cancer detected."
        condition = "Condition: Normal."
    
    return img, diagnosis, condition, probability >= 0.5

def open_doctor_site():
    webbrowser.open("https://www.medibuddy.in/doctors/online-consultation/oncologist")

def open_pharma_site():
    webbrowser.open("https://pharmeasy.in/")

def upload_image():
    file_path = filedialog.askopenfilename(filetypes=[("Image Files", "*.jpeg;*.jpg;*.png")])
    if not file_path:
        return
    img, diagnosis, condition, has_cancer = predict_cancer(file_path)
    img_display = img.resize((300, 300))
    img_tk = ImageTk.PhotoImage(img_display)
    img_label.config(image=img_tk)
    img_label.image = img_tk
    result_label.config(text=f"{diagnosis}\n{condition}")
    if has_cancer:
        consult_doctor_button.pack(pady=10)
        buy_medicine_button.pack(pady=10)
    else:
        consult_doctor_button.pack_forget()
        buy_medicine_button.pack_forget()

root = tk.Tk()
root.title("Cancer Detection System")
root.geometry("600x700")

upload_button = tk.Button(root, text="Upload CT Scan", command=upload_image, font=("Arial", 14), bg="#4CAF50", fg="white")
upload_button.pack(pady=20)

img_label = tk.Label(root)
img_label.pack()

result_label = tk.Label(root, text="", font=("Arial", 14), fg="blue", wraplength=500, justify="left")
result_label.pack(pady=20)

consult_doctor_button = tk.Button(root, text="Consult Oncologist", font=("Arial", 14), bg="#FF5733", fg="white", command=open_doctor_site)
consult_doctor_button.pack_forget()

buy_medicine_button = tk.Button(root, text="Buy Medicine", font=("Arial", 14), bg="#007BFF", fg="white", command=open_pharma_site)
buy_medicine_button.pack_forget()

root.mainloop()
`

**Comment by safwanriyaz** [comment]
here is the video of website with its gui
